authStore.go

```diff
@@ -380,6 +384,42 @@ type sendParams struct {
 	Info             map[string]interface{}
 }
 
+func (s *authStore) RequestPasswordReset(w http.ResponseWriter, r *http.Request, email, templateName, emailSubject string, info map[string]interface{}) error {
+	b := s.b.Clone()
+	defer b.Close()
+	return s.requestPasswordReset(r, b, email, templateName, emailSubject, info)
+}
+
+func (s *authStore) requestPasswordReset(r *http.Request, b Backender, email, templateName, emailSubject string, info map[string]interface{}) error {
+	if !isValidEmail(email) {
+		return newAuthError("Invalid email", nil)
+	}
+
+	u, err := b.GetUser(email)
+	if err != nil {
+		return nil // user does not exist, send success message anyway to prevent fishing for user data
+	}
+
+	if info == nil {
+		info = make(map[string]interface{})
+	}
+	for key := range u.Info {
```

Comment:
You're overwriting the destination URL being passed in here. So, I think you need to do this the opposite way. Start with u.Info and overwrite u.Info with values from info.

```
if info != nil {
    for key, value := range info {
        u.Info[key] = value
    }
}
```

Then use u.Info in the call to addEmailSession